File: app/src/main/java/com/ozhan/mustafa/pokemongoeye/NotificationListener.java
package com.ozhan.mustafa.pokemongoeye;

import android.annotation.TargetApi;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.os.IBinder;
import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;
import android.util.Log;

/**
 * Created by Mustafa Özhan on 7/28/16.
 */
public class NotificationListener extends NotificationListenerService {
    Context context;

    @Override
    public void onCreate() {
        super.onCreate();
        context=getApplicationContext();

    }

    @Override
    public IBinder onBind(Intent intent) {
        return super.onBind(intent);
    }

    @TargetApi(Build.VERSION_CODES.KITKAT)
    @Override
    public void onNotificationPosted(StatusBarNotification sbn) {
        super.onNotificationPosted(sbn);

        String NT_pack=sbn.getPackageName();
        String NT_ticker=sbn.getNotification().tickerText.toString();
        Bundle NT_extras=sbn.getNotification().extras;
        String NT_title=NT_extras.getString("android.title");
        String NT_text=NT_extras.getCharSequence("android.text").toString();
        Log.d("Yeni Bildirim: ", "Package: "+NT_pack+", Title: "+NT_title+", Text: "+NT_text);

    }

    @Override
    public void onNotificationRemoved(StatusBarNotification sbn) {
        super.onNotificationRemoved(sbn);
    }
}

File: app/src/main/java/com/ozhan/mustafa/pokemongoeye/NotificationListener.kt
package com.ozhan.mustafa.pokemongoeye

import android.annotation.SuppressLint
import android.annotation.TargetApi
import android.content.Context
import android.os.Build
import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification
import android.util.Log

@SuppressLint("Registered", "OverrideAbstract")
/**
 * Created by Mustafa Özhan on 7/28/16.
 */
class NotificationListener : NotificationListenerService() {
    private lateinit var context: Context

    override fun onCreate() {
        super.onCreate()
        context = applicationContext

    }

    @TargetApi(Build.VERSION_CODES.KITKAT)
    override fun onNotificationPosted(sbn: StatusBarNotification) {
        super.onNotificationPosted(sbn)

        val ntPack = sbn.packageName
        val ntTicker = sbn.notification.tickerText.toString()
        val ntExtras = sbn.notification.extras
        val ntTitle = ntExtras.getString("android.title")
        val ntText = ntExtras.getCharSequence("android.text")!!.toString()
        Log.d("Yeni Bildirim: ", "Package: $ntPack, Title: $ntTitle, Text: $ntText")

    }

}
